cloud/scope/cluster.go

```diff
@@ -484,25 +637,45 @@ func (s *ClusterScope) ensureLoadBalancerUnique(loadBalancerName string) (*vpcv1
 func (s *ClusterScope) DeleteLoadBalancer() (bool, error) {
 	deleted := false
 	if lbipID := s.GetLoadBalancerID(); lbipID != "" {
-		listLoadBalancersOptions := &vpcv1.ListLoadBalancersOptions{}
-		loadBalancers, _, err := s.IBMVPCClient.ListLoadBalancers(listLoadBalancersOptions)
-		if err != nil {
-			return deleted, err
-		}
-
-		for _, loadBalancer := range loadBalancers.LoadBalancers {
-			if (*loadBalancer.ID) == lbipID {
-				deleted = true
-				if *loadBalancer.ProvisioningStatus != string(infrav1beta2.VPCLoadBalancerStateDeletePending) {
-					deleteLoadBalancerOption := &vpcv1.DeleteLoadBalancerOptions{}
-					deleteLoadBalancerOption.SetID(lbipID)
-					_, err := s.IBMVPCClient.DeleteLoadBalancer(deleteLoadBalancerOption)
-					if err != nil {
-						record.Warnf(s.IBMVPCCluster, "FailedDeleteLoadBalancer", "Failed loadBalancer deletion - %v", err)
-						return deleted, err
+		f := func(start string) (bool, string, error) {
+			// check for existing loadBalancers
+			listLoadBalancersOptions := &vpcv1.ListLoadBalancersOptions{}
+			if start != "" {
+				listLoadBalancersOptions.Start = &start
+			}
+
+			loadBalancersList, _, err := s.IBMVPCClient.ListLoadBalancers(listLoadBalancersOptions)
+			if err != nil {
+				return false, "", err
+			}
+
+			if loadBalancersList == nil {
+				return false, "", fmt.Errorf("loadBalancer list returned is nil")
+			}
+
+			for _, lb := range loadBalancersList.LoadBalancers {
+				if (*lb.ID) == lbipID {
+					deleted = true
+					if *lb.ProvisioningStatus != string(infrav1beta2.VPCLoadBalancerStateDeletePending) {
+						deleteLoadBalancerOption := &vpcv1.DeleteLoadBalancerOptions{}
+						deleteLoadBalancerOption.SetID(lbipID)
+						_, err := s.IBMVPCClient.DeleteLoadBalancer(deleteLoadBalancerOption)
+						if err != nil {
+							record.Warnf(s.IBMVPCCluster, "FailedDeleteLoadBalancer", "Failed loadBalancer deletion - %v", err)
+							return false, "", err
+						}
 					}
 				}
 			}
+
+			if loadBalancersList.Next != nil && *loadBalancersList.Next.Href != "" {
+				return false, *loadBalancersList.Next.Href, nil
+			}
+			return true, "", nil
+		}
+
+		if err := utils.PagingHelper(f); err != nil {
+			return false, err
```

Comment:
Currently true is returned from [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/cloud/scope/cluster.go#L502) in case of en error. Now we are returning false. Is this supposed to be false here? or Am I suspecting wrong?

Reply:
This should be okay as it is with false, anyways the error check will always come in first  before the boolean evaluation.

Reply:
ok makes sense!